src/Microsoft.NetCore.Analyzers/Core/Runtime/SerializationRulesDiagnosticAnalyzer.cs

```diff
@@ -128,7 +128,8 @@ public override void Initialize(AnalysisContext analysisContext)
                         return;
                     }
 
-                    var isNetStandardAssembly = context.Compilation.ReferencedAssemblyNames.Any(identity => string.Equals(identity.Name, "netstandard", StringComparison.OrdinalIgnoreCase));
+                    var systemObjectSymbol = context.Compilation.GetSpecialType(SpecialType.System_Object);
+                    var isNetStandardAssembly = systemObjectSymbol.ContainingAssembly.Name != "mscorlib";
```

Comment:
Can we make this an extension method on Compilation? Can be done in a separate PR...

Reply:
Heh, I commented on this right before you did 👍